pkg/ottl/contexts/ottldatapoints/datapoints.go

```diff
@@ -434,17 +431,17 @@ func accessTimeUnixNano() ottl.StandardGetSetter[TransformContext] {
 func accessDoubleValue() ottl.StandardGetSetter[TransformContext] {
 	return ottl.StandardGetSetter[TransformContext]{
 		Getter: func(ctx TransformContext) interface{} {
-			switch ctx.GetDataPoint().(type) {
-			case pmetric.NumberDataPoint:
-				return ctx.GetDataPoint().(pmetric.NumberDataPoint).DoubleValue()
+			numberDataPoint, ok := ctx.GetDataPoint().(pmetric.NumberDataPoint)
```

Comment:
```suggestion
			if numberDataPoint, ok := ctx.GetDataPoint().(pmetric.NumberDataPoint); ok {
```

I think keeping the assignment in the `if` statement would be a bit more succinct.

Reply:
Youre right. I should stop multitasking

Reply:
I do think it is more succinct but harder to read. It is much easier to follow the code without having to go to the "branches".